backends/bmv2/bmv2.cpp

```diff
@@ -49,22 +49,38 @@ int main(int argc, char *const argv[]) {
     auto program = parseP4File(options);
     if (program == nullptr || ::errorCount() > 0)
         return 1;
-    P4::FrontEnd frontend;
-    frontend.addDebugHook(hook);
-    program = frontend.run(options, program);
+    try {
+        P4::FrontEnd frontend;
+        frontend.addDebugHook(hook);
+        program = frontend.run(options, program);
+    } catch (const Util::P4CExceptionBase &bug) {
```

Comment:
In theory for debugging purposes you could use `catch throw` in gdb. I haven't ever actually tried that, though, so I'm not sure how well it works in practice.